### PR TITLE
Fix exceptions thrown in a run config breaking the Run Config action …

### PR DESCRIPTION
**Pull request by abrooksv**
…button

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
* Resolve some TODOs to simplify the LambdaBuilder logic
* Make sure all exceptions get mapped into CompletableFuture.completeExceptionally

## Related Issue(s)
#868

## Testing
Forced exceptions to be thrown, action button still works after

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt [hunk 1]
```diff
@@ -133,7 +133,11 @@ abstract class LambdaBuilder {
             processHandler.startNotify()
         }
 
-        return future
+        return try {
+            future.get()
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: Its used in an async method in LambdaBuilderUtils, which has to have this call block, else the progress bar immediately goes away

[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: https://github.com/aws/aws-toolkit-jetbrains/pull/896/files#diff-af40ca3afd894931a84a159370b1357fR91

[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: Discussed offline, we need this method to block till underlying sam process is done. Removed the unneeded run on BG thread call though

**Comment by abrooksv** [comment]
@aws/aws-ides-team 	Any other concerns?

**Comment by zhangzhx** [comment]
LGTM. We have the `LambdaBuilder` file refactored a bit, need to resolve conflicts 😬

**Pull request merged**